app/forms/two_factor_authentication/phone_deletion_form.rb

```diff
@@ -0,0 +1,52 @@
+module TwoFactorAuthentication
+  class PhoneDeletionForm
+    include ActiveModel::Model
+
+    attr_reader :user, :configuration
+
+    def initialize(user, configuration)
+      @user = user
+      @configuration = configuration
+    end
+
+    def submit
+      success = configuration_absent? ||
+                configuration_owned_by_user? && multiple_factors_enabled? && configuration_destroyed
+
+      FormResponse.new(success: success, errors: errors.messages, extra: extra_analytics_attributes)
+    end
+
+    private
+
+    def extra_analytics_attributes
+      {
+        user_id: user.uuid,
```

Comment:
This isn't necessary. It's tracked automatically for authenticated users.

Reply:
Ahh. Okay. I had seen this elsewhere, so cargo-culted it.

Reply:
It's only needed when the user is not yet authenticated, but we can find out who they are based on a token, for example, when confirming email, resetting password, etc.